custom_model_runner/datarobot_drum/resource/predict_mixin.py

```diff
@@ -53,10 +54,21 @@ def do_predict(self, logger=None):
         with tempfile.NamedTemporaryFile(suffix=file_ext) as f:
             filestorage.save(f)
             f.flush()
-            out_data = self._predictor.predict(f.name)
+            if self._target_type == TargetType.TRANSFORM.value:
+                out_data = self._predictor.transform(f.name)
+            else:
+                out_data = self._predictor.predict(f.name)
 
-        if self._target_type == TargetType.UNSTRUCTURED:
+        if self._target_type == TargetType.UNSTRUCTURED.value:
```

Comment:
TargetType.UNSTRUCTURED should not take value.
self._target_type is enum, not str, check it's assignment in `BaseLanguagePredictor`

And the same for TRANSFORM further in the code

Reply:
I must be doing something wrong in my test then- I get a mismatch when I don't check the .value for some reason